File: Python/Counting_sort.py
def countingSort(array):
    n = len(array)
    count = [0] * (max(array) + 1)
    answer = [0] * n
    for i in range(n):
        count[array[i]] += 1
    for i in range(1, len(count)):
        count[i] += count[i-1]
    for i in range(n-1, -1, -1):
        count[array[i]] -= 1
        answer[count[array[i]]] = array[i]
    return answer


arr = [5, 8, 9, 2, 1, 2, 2, 3, 4, 5, 6, 7, 9, 10, 0, 0]
print(countingSort(arr))

File: Java/counting_sort.java
import java.util.Random;

public class counting_sort {
    public int[] countingSort(int []a){
        int [] count,b;
        count = new int[10000];
        b = new int[10000];
        for(int i=0;i<10000;i++)
            count[a[i]]++;
        for(int i=1;i<10000;i++)
            count[i]+=count[i-1];
        for(int i=a.length-1;i>=0;i--){
            b[count[a[i]]-1] = a[i];
            count[a[i]]--;
        }
        return b;
    }
    public static void main(String[] args) {
        Random rand = new Random();
        counting_sort obj = new counting_sort();
        int[] a = new int[10000];
        for(int i=0;i<10000;i++)
            a[i]= rand.nextInt(10000);
        long startTime = System.nanoTime();
        a = obj.countingSort(a);
        long endTime = System.nanoTime();
        System.out.println("Time taken = "+(endTime-startTime)/1000);
    }
}